links/site/src/index.js

```diff
@@ -0,0 +1,237 @@
+import React from 'react';
+import ReactDOM from 'react-dom';
+import { BrowserRouter, Route, Switch } from 'react-router-dom'
+import * as firebase from 'firebase/app'
+import 'firebase/auth'
+import 'firebase/database'
+import config from './config'
+
+firebase.initializeApp(config);
+
+class App extends React.Component {
+
+    constructor(props) {
+        super(props)
+        this.state = {
+            isAuthed: false
+        }
+    }
+
+    render() {
+        console.log('render app', this.state)
+        if (this.state.isAuthed) {
+            return (
+                <BrowserRouter>
+                    <Switch>
+                        <Route exact path="/*" component={Editor} />
+                    </Switch>
+
+                </BrowserRouter>
+            )
+        } else {
+            return (
+                <BrowserRouter>
+                    <Switch>
+                        <Route path="/*" component={AuthComponent} />
+                    </Switch>
+                </BrowserRouter>
+            )
+        }
+    }
+
+    componentDidMount() {
+        this.unsubscribe = firebase.auth().onAuthStateChanged(user => {
+            this.setState({
+                isAuthed: Boolean(user) && !user.isAnonymous
+            })
+        })
+    }
+
+    componentWillUnmount() {
+        this.unsubscribe()
```

Comment:
does it matter at this point to change the state?

Reply:
react warns about memory leaks due to an async which may not have finished when unmounting